src/hooks/use-ordered-rows.ts

```diff
@@ -2,29 +2,44 @@ import { useMemo } from 'preact/hooks';
 
 import type { Order } from '../types';
 
+export type UseOrderedRowsOptions = {
+  /**
+   * Make case-sensitive comparisons for string values.
+   * Defaults to false.
+   */
+  caseSensitive?: boolean;
+};
+
 /**
  * Orders a list of rows based on provided order options.
  * Provided rows are not mutated, but a copy is returned instead.
+ * Values are compared preserving their type.
  */
 export function useOrderedRows<Row>(
   rows: Row[],
   order?: Order<keyof Row>,
+  { caseSensitive = false }: UseOrderedRowsOptions = {},
 ): Row[] {
   return useMemo(() => {
     if (!order) {
       return rows;
     }
 
-    return [...rows].sort((a, b) => {
-      if (a[order.field] === b[order.field]) {
+    return [...rows].sort(({ [order.field]: a }, { [order.field]: b }) => {
+      const aField =
+        !caseSensitive && typeof a === 'string' ? a.toLowerCase() : a;
+      const bField =
+        !caseSensitive && typeof b === 'string' ? b.toLowerCase() : b;
```

Comment:
This handles ASCII but doesn't deal with things like accented characters. With this implementation, "José" would sort after "Joshua" for example. One way to resolve this would be to build on `Intl.Collator`. That has a `sensitivity` option amongst others.

Reply:
Hah, good point. Let me take a look at `Intl.Collator`.